core/blockchain.go

```diff
@@ -1211,7 +1211,7 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 		// * ancient-limit: the indices of blocks before ancient-limit are ignored
 		if tail := rawdb.ReadTxIndexTail(bc.db); tail == nil {
 			if bc.txLookupLimit == 0 || ancientLimit <= bc.txLookupLimit {
-				rawdb.WriteTxIndexTail(bc.db, 0)
+				//rawdb.WriteTxIndexTail(bc.db, 0)
```

Comment:
Any particular reason for removing it?

During the snap sync, all blocks in the scope of index range will be indexed directly and just mark the tail flag here to avoid duplicated work.

Reply:
> Any particular reason for removing it?
> 
> During the snap sync, all blocks in the scope of index range will be indexed directly and just mark the tail flag here to avoid duplicated work.

If crashing during reindex, the tail will be set 0 first, but the index will not be completed.